TripKitUI/controller/TKSectionedAlertViewModel.swift

```diff
@@ -0,0 +1,151 @@
+//
+//  TKSectionedAlertViewModel.swift
+//  TripKitUI-iOS
+//
+//  Created by Kuan Lun Huang on 15/3/18.
+//  Copyright © 2018 SkedGo. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+import RxDataSources
+
+public class TKSectionedAlertViewModel {
+  
+  public let region: SVKRegion
+  
+  private let disposeBag = DisposeBag()
+  
+  lazy var sections: Observable<[AlertSection]> = {
+    return TKBuzzInfoProvider.rx_fetchTransitAlerts(forRegion: region)
+      .map { self.alertItems(from: $0) }
+      .map { self.alertSections(from: $0) }
+      .share(replay: 1)
+  }()
+  
+  public init(region: SVKRegion) {
+    self.region = region
+  }
+  
+  // MARK: -
+  
+  private func alertItems(from decoded: [API.AlertMapping]) -> [String: [AlertGroup]] {
```

Comment:
This is still WIP and requires clean up (suggestion welcome). The purpose of this method is to group alerts by some route id, whether it's `id`, `name` or `number`.